### ISSUE TITLE
Full system diagnostic report

### ISSUE THREAD

**Issue by nblumhardt** [datalust/seq-releases#594]
It's often necessary to collect a variety of server information for support purposes. Rather than have customers spend significant time on this, Seq should provide an all-in-one diagnostic report with absolutely everything included.

**Comment by nblumhardt** [comment]
![image](https://user-images.githubusercontent.com/342712/27848054-82c881c2-6185-11e7-952d-79c9acd20595.png)

**Comment by nblumhardt** [comment]
Below is the initial version of the report. Some sections won't be able to be generated if Seq is running unprivileged, so in these cases an error will be printed in the corresponding section.

```
====================
Seq Diagnostic Report 2017-07-05T03:34:51.8781192Z
--------------------
 1. Executable
 2. Operating System
 3. Processor #1
 4. System Memory
 5. Process Memory
 6. Process
 7. System Storage
 8. Event Storage
 9. Event RAM Cache
 10. Configuration
 11. License Features
 12. Document Store
 13. Apps
 14. Activity
 15. Ingestion Endpoint
 16. Query Cache
 17. Ingestion Log
 18. Server Log (up to 5000 entries)

====================
1. Executable
--------------------
Seq Version         : 1.0.0-pre
Bitness             : 64
Framework Version   : 4.0.30319.42000

====================
2. Operating System
--------------------
OS Caption          : Microsoft Windows 10 Pro
OS Version          : 10.0.15063
OS Bitness          : 64

====================
3. Processor #1
--------------------
Cores               : 4
Enabled Cores       : 4
Logical Processors  : 8
Maximum Clock Speed : 2808
Caption             : Intel64 Family 6 Model 158 Stepping 9

====================
4. System Memory
--------------------
Physical Total      : 17008504832
Physical Available  : 9721749504
Physical Utilization: 0.428418335413623
Commit Limit        : 30440128512
Commit Peak         : 31457357824
Commit Total        : 26565431296
Process Count       : 210
Handle Count        : 83734
Thread Count        : 2462
Kernel Total        : 745115648
Kernel Paged        : 466849792
Kernel Non-paged    : 278265856
Page Size           : 4096
System Cache        : 3987652608

====================
5. Process Memory
--------------------
Working Set (Environment): 1069993984
Working Set         : 1069993984
Paged               : 19449487360
Paged System        : 17969220240
Non-paged System    : 730130
Private             : 19449487360
Handle Count        : 1084
Thread Count        : 57

====================
6. Process
--------------------
Uptime              : 00:00:22.1220211
Thread Pool Threads Available: 32764
Thread Pool IOCP Threads Available: 999

====================
7. System Storage
--------------------
Space Remaining     : 374483140608

====================
8. Event Storage
--------------------
First Extent Range Start: 2015-05-06T00:00:00.0000000Z
Last Extent Range End: 2017-07-12T00:00:00.0000000Z

====================
9. Event RAM Cache
--------------------
Days Cached         : 128
Events Cached       : 392229
Is Warming Up       : Yes

====================
10. Configuration
--------------------
Preferred Storage Engine: LMDB
LMDB Extent Map Size: 200000000000
System Memory Utilization Target: 0.9
Minimum Free Disk Space: 134217728
Lazy Flush Enabled  : No
Raw Event Maximium Content Length: 262144
Raw Payload Maximum Content Length: 10485760
API Keys Required   : No
Authentication Enabled: Yes
Active Directory Authentication: No
Azure Active Directory Authentication: No

====================
11. License Features
--------------------
Serial Number       : SDES-01-3855202
Licensed User Count : <null>
Includes Active Directory: Yes

====================
12. Document Store
--------------------
Active Sessions     : 1

====================
13. Apps
--------------------
Running Apps Total  : 3
Email+ (Seq.App.EmailPlus v2.0.57) Instances: 2
File Archive (Seq.App.FileArchive v2.0.44) Instances: 1

====================
14. Activity
--------------------
Task "Sample system memory" Running Time: 00:00:20.8270447
Task "Manage event cache resources" Running Time: 00:00:20.8270447
Task "Monitor alerts and trigger notifications" Running Time: 00:00:20.6637962

====================
15. Ingestion Endpoint
--------------------
Events Arrived per Minute: 0
Events Accepted per Minute: 0
Ingested Data per Minute: 0
Invalid Payloads per Minute: 0
Unauthorized Payloads per Minute: 0

====================
16. Query Cache
--------------------
Queries Executed per Minute: 0
Time Slice Hits per Minute: 0
Invalidations per Minute: 0

====================
17. Ingestion Log
--------------------
{"@t":"2017-07-05T03:34:29.7799378Z","@mt":"The ingestion API is configured and accepting requests","@l":"Debug"}

====================
18. Server Log (up to 5000 entries)
--------------------
{"@t":"2017-07-05T03:34:49.2224417Z","@mt":"Deduplication buffer renewed on reaching {@Shared}","Shared":{"Filter":0.016390085220336914,"Schemata":94,"Strings":102968,"Quality":7},"SourceContext":"Flare.Events.Cache.SegmentCache","OperationId":"e1a065cb-fa83-4e35-a11f-cb4b8ab22536"}
{"@t":"2017-07-05T03:34:30.9360741Z","@mt":"Cache warm-up is required","SourceContext":"Flare.Events.Cache.SegmentCache","OperationId":"e1a065cb-fa83-4e35-a11f-cb4b8ab22536"}
{"@t":"2017-07-05T03:34:30.9195763Z","@mt":"Storage subsystem available"}
{"@t":"2017-07-05T03:34:30.7866642Z","@mt":"Available storage engines in order of preference are {Engines}","Engines":["LMDB","ESENT"],"SourceContext":"Flare.Events.EventStore"}
{"@t":"2017-07-05T03:34:30.7841862Z","@mt":"Opening event store at {ExtentsPath}","ExtentsPath":"C:\\ProgramData\\Seq\\Extents","SourceContext":"Flare.Events.EventStore"}
{"@t":"2017-07-05T03:34:30.3544976Z","@mt":"ESENT sparse file support is available on this system"}
{"@t":"2017-07-05T03:34:30.0198634Z","@mt":"Opening document store {DatabasePath}","DatabasePath":"C:\\ProgramData\\Seq\\Documents\\default.quince","InstanceName":"Quince","SourceContext":"Quince.QuinceDB"}
{"@t":"2017-07-05T03:34:29.9754864Z","@mt":"Seq listening on {ListenUris}","ListenUris":["http://localhost:5341/"]}
{"@t":"2017-07-05T03:34:29.5562453Z","@mt":"Seq {SeqVersion} {ProcessWordSize}-bit running on .NET {DotNetVersion} and OS {OSCaption} ({OSVersion}) {OSWordSize}-bit","SeqVersion":"1.0.0-pre","ProcessWordSize":64,"DotNetVersion":"4.0.30319.42000","OSCaption":"Microsoft Windows 10 Pro","OSVersion":"10.0.15063","OSWordSize":64,"SourceContext":"Seq.Server.Features.Diagnostics.EnvironmentInformationWriter"}
```
